inbc-program/inbc/parser/source_app_parser.py

```diff
@@ -47,13 +56,16 @@ def application_remove(args: argparse.Namespace) -> str:
     manifest = ('<?xml version="1.0" encoding="utf-8"?>' +
                 '<manifest><type>source</type>' +
                 '<applicationSource>' +
-                '<remove><gpg>'
-                f'{create_xml_tag(arguments, "keyname")}' +
-                '</gpg><repo>' +
-                f'{create_xml_tag(arguments, "filename")}'
-                '</repo>'
-                '</remove></applicationSource>' +
-                '</manifest>')
+                '<remove>')
+
+    if args.gpgKeyName:
+        manifest += f'<gpg>{create_xml_tag(arguments, "keyname")}</gpg>'
+
+    manifest += ('<repo>' +
+                 f'{create_xml_tag(arguments, "filename")}'
+                 '</repo>'
+                 '</remove></applicationSource>' +
+                 '</manifest>')
 
     print(f"manifest {manifest}")
```

Comment:
**Suggestion:** Use a context manager to handle the printing of the manifest to ensure that resources are properly managed. [best practice]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    with open('manifest_log.txt', 'a') as log_file:
        log_file.write(f"manifest {manifest}\n")
```